sql-jdbc/src/TableauConnector/opensearch_sql_jdbc/dialect.tdd

```diff
@@ -63,16 +63,87 @@
             <argument type='datetime' />
             <argument type='datetime' />
         </function>
+
+        <function group='operator' name='==' return-type='bool'>
+            <formula>(TIMESTAMP(%1) = TIMESTAMP(%2))</formula>
+            <argument type='date' />
+            <argument type='datetime' />
+        </function>
+        <function group='operator' name='==' return-type='bool'>
+            <formula>(TIMESTAMP(%1) = TIMESTAMP(%2))</formula>
+            <argument type='datetime' />
```

Comment:
I think just calling TIMESTAMP would cause an issue where the time would not be 00:00:00 as expected. We have to do something like SUBDATE(TIMESTAMP(DATE(%2)), INTERVAL HOUR(TIMESTAMP(DATE(%2))) HOUR). This will likely result in a an error logged in this [issue](https://github.com/opensearch-project/sql/issues/294).

Reply:
It works fine, see the test on the pic:
![image](https://user-images.githubusercontent.com/88679692/142672056-34eeabd3-d2af-4ae6-be60-044e3d5a726f.png)